lib/matplotlib/backends/backend_nbagg.py

```diff
@@ -96,6 +96,15 @@ def show(self):
         else:
             self.canvas.draw_idle()
         self._shown = True
+        # plt.figure adds an event which makes the figure in focus the active
+        # one. Disable this behaviour, as it results in figures being put as
+        # the active figure after they have been shown, even in non-interactive
+        # mode.
+        if hasattr(self, '_cidgcf'):
```

Comment:
Is there any way to add tests to argue that this is working as expected?

Reply:
Actually, there was previously no test coverage for the corresponding old code paths (as can be checked e.g. via `pytest --cov --pyargs matplotlib.tests.test_backend_nbagg && coverage html`: the entire show() function was untested), so I'll ask for an out there and just point out that I simply moved some code around (also, my understanding of nbagg is on the weaker side...).

Reply:
fair.